editor/src/messages/portfolio/document/utility_types/transformation.rs

```diff
@@ -1,14 +1,34 @@
 use crate::consts::{ROTATE_SNAP_ANGLE, SCALE_SNAP_INTERVAL};
+use crate::messages::portfolio::document::node_graph::VectorDataModification;
 use crate::messages::prelude::*;
-
+use crate::messages::tool::common_functionality::shape_editor::ShapeState;
+use crate::messages::tool::utility_types::ToolType;
 use document_legacy::document::Document;
 use document_legacy::layers::style::RenderData;
 use document_legacy::LayerId;
+use graphene_core::vector::{ManipulatorPointId, SelectedType};
 
 use glam::{DAffine2, DVec2};
 use std::collections::{HashMap, VecDeque};
 
-pub type OriginalTransforms = HashMap<Vec<LayerId>, DAffine2>;
+#[derive(Debug, PartialEq, Clone)]
+pub enum OriginalTransforms {
+	Layer(HashMap<Vec<LayerId>, DAffine2>),
+	Path(HashMap<Vec<LayerId>, Vec<(ManipulatorPointId, DVec2)>>),
+}
+impl Default for OriginalTransforms {
+	fn default() -> Self {
+		OriginalTransforms::Path(HashMap::new())
+	}
+}
```

Comment:
```suggestion
#[derive(Debug, Default, PartialEq, Clone)]
pub enum OriginalTransforms {
	Layer(HashMap<Vec<LayerId>, DAffine2>),
	#[default]
	Path(HashMap<Vec<LayerId>, Vec<(ManipulatorPointId, DVec2)>>),
}

```

Reply:
@Keavon This is only available for unit variants.

Your code will not compile.